feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/usecases/CreatePersonEventUseCase.kt

```diff
@@ -27,70 +23,53 @@ internal class CreatePersonEventUseCase @Inject constructor(
         // If a personCreationEvent is already in the current session,
         // we don' want to add it again (the capture steps would still be the same)
         if (sessionEvents.none { it is PersonCreationEvent }) {
-            val faceCaptureBiometricsEvents = sessionEvents.filterIsInstance<FaceCaptureBiometricsEvent>()
-            val fingerprintCaptureBiometricsEvents = sessionEvents.filterIsInstance<FingerprintCaptureBiometricsEvent>()
+            val faceCaptures = extractFaceCaptures(results)
+            val fingerprintCaptures = extractFingerprintCaptures(results)
 
-            val faceSamples = extractFaceSamples(results)
-            val fingerprintSamples = extractFingerprintSamples(results)
+            val personCreationEvent = build(faceCaptures, fingerprintCaptures)
 
-            val personCreationEvent = build(
-                faceCaptureBiometricsEvents,
-                fingerprintCaptureBiometricsEvents,
-                faceSamples,
-                fingerprintSamples
-            )
             if (personCreationEvent.hasBiometricData()) {
```

Comment:
Should we log an error in the opposite case?

Reply:
We could, but I am pretty sceptical that anyone will ever look into it.